### PR TITLE
add ldap managment

### PR DESCRIPTION
**Pull request by b4ldr**

### CODE REVIEW

File: types/group_mapping.pp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type Grafana::Group_mapping = Struct[{
```
[the remark this answers is not in the document]
Comment (b4ldr) [reply types/group_mapping.pp]: if the change above is made would you then want it to be `Grafana::Settings::Auth_ldap:: Group_mapping`

As top CamelCase im not sure stdlib does use that standard.  i cant see any examples of that however the compact section does have examples of underscore[1].  Further using underscore is much more inline with the puppet style guide[3]

[1]https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/compat/ip_address.pp
[2]https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/compat/absolute_path.pp

File: types/server_attributes.pp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type Grafana::Server_attributes = Struct[{
+  name => String[1],
+  surename => String[1],
```
[the remark this answers is not in the document]
Comment (b4ldr) [reply types/server_attributes.pp]: thanks, fixed

**Comment by b4ldr** [comment]
happy to rebase once params have gone if you just ping this PR at that time

File: manifests/settings/auth_ldap.pp [hunk 3]
```diff
@@ -10,10 +10,5 @@
 
   grafana::settings { 'auth.ldap': settings => $settings }
 
-  file { $config_file:
-    ensure => file,
-    owner  => $grafana::user,
-    group  => $grafana::group,
-    mode   => '0640',
-  }
+  include grafana::ldap
```
[the remark this answers is not in the document]
Comment (johanfleury) [reply manifests/settings/auth_ldap.pp]: Yes, that's what I had in mind.

File: types/group_mapping.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanfleury) [reply types/group_mapping.pp]: > if the change above is made would you then want it to be `Grafana::Settings::Auth_ldap::Group_mapping`

Yes please.

> using underscore is much more inline with the puppet style guide[3]

I agree with that. Forget about my previous comment, let's use underscore :+1:

**Comment by johanfleury** [comment]
Sorry for closing the PR, it wasn't intentional.

> happy to rebase once params have gone if you just ping this PR at that time

Done :)

**Comment by b4ldr** [comment]
All changes should be done now and running the new code in my environment

File: data/common.yaml [hunk 4]
```diff
@@ -11,3 +10,21 @@ grafana::config_file: "%{lookup('grafana::config_dir')}/grafana.ini"
 
 grafana::user: grafana
 grafana::group: grafana
+grafana::settings::auth_ldap::ldap_servers:
+  - localhost
+grafana::settings::auth_ldap::enabled: true
```
Comment (johanfleury) [inline data/common.yaml]: Sorry for not seeing this before but I think LDAP authentication should be disabled by default.

Comment (b4ldr) [reply data/common.yaml]: My thinking hear is that by default ldap wont be enabled.  it is only enabled if someone calls

```puppet
class grafana {
  settings => {
    'auth.ldap' => { ...}
  }
}
```
or `class { 'grafana::settings::auth_ldap': }` directly if one considers this then i think a default of enabled makes more senses but happy to change.

Comment (johanfleury) [reply data/common.yaml]: You're right, once again.

**Comment by johanfleury** [comment]
Thank you again for your work @b4ldr.

**Pull request merged**